### PR TITLE
** DO NOT MERGE **  Priority queue pattern doc update

### PR DESCRIPTION
**Pull request by magrande**

### CODE REVIEW

**Comment by PRMerger-2** [comment]
@magrande : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/patterns/priority-queue.md [hunk 1]
```diff
@@ -87,74 +87,47 @@ In the figure above, the application creates several messages and assigns a cust
...
-  }
-}
+public static class PriorityQueueConsumerHighFn
+    {
+        [FunctionName("HighPriorityQueueConsumerFunction")]
+        public static void Run([ServiceBusTrigger("messages", "highPriority", Connection = "ServiceBusConnection")]string highPriorityMessage, ILogger log)
+        {
+            log.LogInformation($"C# ServiceBus topic trigger function processed message: {highPriorityMessage}");
+        }
+    }
```
Comment (ckittel) [inline docs/patterns/priority-queue.md:100]:
```suggestion
{
    [FunctionName("HighPriorityQueueConsumerFunction")]
    public static void Run([ServiceBusTrigger("messages", "highPriority", Connection = "ServiceBusConnection")]string highPriorityMessage, ILogger log)
    {
        log.LogInformation($"C# ServiceBus topic trigger function processed message: {highPriorityMessage}");
    }
}
```

File: docs/patterns/priority-queue.md [hunk 2]
```diff
@@ -87,74 +87,47 @@ In the figure above, the application creates several messages and assigns a cust
...
-```csharp
-protected override async Task ProcessMessage(BrokeredMessage message)
-{
-  // Simulate message processing for High priority messages.
-  await base.ProcessMessage(message);
-  Trace.TraceInformation("High priority message processed by " +
-    RoleEnvironment.CurrentRoleInstance.Id + " MessageId: " + message.MessageId);
-}
-```
+An administrator can configure how many instances the functions on the app service consumption can scale out to by setting a value to the WEBSITE_MAX_DYNAMIC_APPLICATION_SCALE_OUT application setting. Typically there'll be more instances of the `PriorityQueueConsumerHigh` function than the `PriorityQueueConsumerLow` function.
```
Comment (ckittel) [inline docs/patterns/priority-queue.md:103]:
```suggestion
An administrator can configure how many instances the functions on the app service consumption can scale out to by setting a value to the [`WEBSITE_MAX_DYNAMIC_APPLICATION_SCALE_OUT`](https://docs.microsoft.com/azure/azure-functions/functions-app-settings#website_max_dynamic_application_scale_out) application setting. Typically, there'll be more instances of the `PriorityQueueConsumerHigh` function than the `PriorityQueueConsumerLow` function.
```

File: docs/patterns/priority-queue.md [hunk 3]
```diff
@@ -87,74 +87,47 @@ In the figure above, the application creates several messages and assigns a cust
...
 
-When an application posts messages to the topic associated with the subscriptions used by the `PriorityQueue.High` and `PriorityQueue.Low` worker roles, it specifies the priority by using the `Priority` custom property, as shown in the following code example. This code (implemented in the `WorkerRole` class in the PriorityQueue.Sender project), uses the `SendBatchAsync` helper method of the `QueueManager` class to post messages to a topic in batches.
+Another project named `PriorityQueueSender` contains a time triggered Azure Function configured to run every 30 seconds; this Azure Function integrates with Azure Service Bus via an output binding and sends batches of low and high priority messages to an IAsyncCollector Topic; when the function posts messages to the topic associated with the subscriptions used by the `PriorityQueueConsumerHigh` and `PriorityQueueConsumerLow` functions, it specifies the priority by using the `Priority` custom property, as shown in the following code:
```
Comment (ckittel) [inline docs/patterns/priority-queue.md:105]:
```suggestion
Another project named `PriorityQueueSender` contains a time triggered Azure Function configured to run every 30 seconds; this Azure Function integrates with Azure Service Bus via an output binding and sends batches of low and high priority messages to an `IAsyncCollector`; when the function posts messages to the topic associated with the subscriptions used by the `PriorityQueueConsumerHigh` and `PriorityQueueConsumerLow` functions, it specifies the priority by using the `Priority` custom property, as shown in the following code:
```

File: docs/patterns/priority-queue.md [hunk 4]
```diff
@@ -87,74 +87,47 @@ In the figure above, the application creates several messages and assigns a cust
...
+                await collector.AddAsync(lpMessage);
+
+                messageId = Guid.NewGuid().ToString();
+                var hpMessage = new ServiceBusMessage() { MessageId = messageId };
+                hpMessage.ApplicationProperties["Priority"] = Priority.High;
+                hpMessage.Body = BinaryData.FromString($"High priority message with Id: {messageId}");
+                await collector.AddAsync(hpMessage);
+            }
+        }
+    }
```
Comment (ckittel) [inline docs/patterns/priority-queue.md:130]:
```suggestion
{
    [FunctionName("PriorityQueueSenderFunction")]
    public static async Task Run(
        [TimerTrigger("0,30 * * * * *")] TimerInfo myTimer,
        [ServiceBus("messages", Connection = "ServiceBusConnection")] IAsyncCollector<ServiceBusMessage> collector)
    {
        for (int i = 0; i < 10; i++)
        {
            var messageId = Guid.NewGuid().ToString();
            var lpMessage = new ServiceBusMessage() { MessageId = messageId };
            lpMessage.ApplicationProperties["Priority"] = Priority.Low;
            lpMessage.Body = BinaryData.FromString($"Low priority message with Id: {messageId}");
            await collector.AddAsync(lpMessage);

            messageId = Guid.NewGuid().ToString();
            var hpMessage = new ServiceBusMessage() { MessageId = messageId };
            hpMessage.ApplicationProperties["Priority"] = Priority.High;
            hpMessage.Body = BinaryData.FromString($"High priority message with Id: {messageId}");
            await collector.AddAsync(hpMessage);
        }
    }
}
```

**Review by ckittel — CHANGES_REQUESTED** [review]
Couple of suggestions

File: docs/patterns/priority-queue.md (hunk 2 above)
Comment (ckittel) [reply docs/patterns/priority-queue.md:103]: But more importantly, this setting is deprecated.  Please update to use: https://docs.microsoft.com/azure/azure-functions/event-driven-scaling#limit-scale-out

**Comment by PRMerger15** [comment]
@magrande : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger19** [comment]
@magrande : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger9** [comment]
@magrande : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/patterns/priority-queue.md (hunk 2 above)
Comment (ckittel) [reply docs/patterns/priority-queue.md]: @magrande -- just wanted to make sure you saw this above.

Comment (magrande) [reply docs/patterns/priority-queue.md]: @ckittel I will take care of this in the samples repo, thanks !

**Comment by PRMerger-2** [comment]
@magrande : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/patterns/priority-queue.md (hunk 2 above)
Comment (ckittel) [reply docs/patterns/priority-queue.md]: Thanks. The new wording looks good.  I'll move this PR over to the internal repo, and will close this one out when complete.

**Review by ckittel — APPROVED** [review]
Moving to internal repo.

**Comment by ckittel** [comment]
Closing this PR as it has been reopened @ MicrosoftDocs/architecture-center-pr#5014

**Pull request closed without merging**